### PR TITLE
Remove oauth provider

### PR DESCRIPTION
**Pull request by gravesm**
##### SUMMARY

This removes the oauth provider functionality from awx. The oauth2_provider app and all references to it have been removed. Migrations to delete the two tables that locally overwrote oauth2_provider tables are included. This change does not include migrations to delete the tables provided by the oauth2_provider app.

##### ISSUE TYPE

 - Breaking Change 
##### COMPONENT NAME

 - API
 - Docs

##### AWX VERSION

```

```

##### ADDITIONAL INFORMATION

```

```

### CODE REVIEW

**Comment by adrisala** [comment]
Shouldn't we also modify references of OAuth usage inside the `awx_collection` directory? The default login method for the collections right now seems to be OAuth.
cc @tvo318 
https://github.com/ansible/awx/blob/devel/awx_collection/README.md#running
https://github.com/ansible/awx/blob/devel/awx_collection/TESTING.md

**Comment by gravesm** [comment]
> Shouldn't we also modify references of OAuth usage inside the `awx_collection` directory? The default login method for the collections right now seems to be OAuth.

Docs will definitely need to be changed. https://github.com/ansible/awx/pull/15554 should have fixed the collection to allow basic auth. I'm still working through test failures, though, so there may be further changes needed.

**Comment by tvo318** [comment]
@adrisala @gravesm Yes I agree, the docs in the `awx_collection` need to be updated as well. Thanks for working on this!

**Comment by chrismeyersfsu** [comment]
I reviewed with an eye towards migrations. If we missed some code to delete or deleted too much we can fix that easily. An error is in the data is harder to retroactivity fix.

I found changes in the activity stream entries. I don't think I care.

## Activity Stream

### Steps to Recreate
* `docker volume rm tools_awx_db_15` <-- blow away your db and start the dev env from the `devel` branch
* Make an application via the API
* Make a token for the application ^^
* Edit the token
* Edit the application
* now swap to this branch and start the dev env (this will run the migrations)
* You now have some activity stream entries

I'll note how those activity stream entries change because of this PR.

<details><summary>App Creation - Before</summary>

```
{
    "id": 21,
    "type": "activity_stream",
    "url": "[/api/v2/activity_stream/21/](http://localhost:8013/api/v2/activity_stream/21/)",
    "related": {
        "actor": "[/api/v2/users/1/](http://localhost:8013/api/v2/users/1/)",
        "o_auth2_application": [
            "[/api/v2/applications/1/](http://localhost:8013/api/v2/applications/1/)"
        ]
    },
    "summary_fields": {
        "o_auth2_application": [
            {
                "id": 1,
                "name": "cmeyers app",
                "description": ""
            }
        ],
        "actor": {
            "id": 1,
            "username": "admin",
            "first_name": "",
            "last_name": ""
        }
    },
    "timestamp": "2024-11-18T13:17:56.321841Z",
    "operation": "create",
    "changes": {
        "name": "cmeyers app",
        "description": "",
        "client_type": "public",
        "redirect_uris": "",
        "authorization_grant_type": "password",
        "skip_authorization": false,
        "organization": "Default-1",
        "id": 1
    },
    "object1": "o_auth2_application",
    "object2": "",
    "object_association": "",
    "action_node": "awx-1",
    "object_type": ""
}
```

</details>

<details><summary>App Creation - After</summary>

```
{
    "id": 21,
    "type": "activity_stream",
    "url": "[/api/v2/activity_stream/21/](http://localhost:8013/api/v2/activity_stream/21/)",
    "related": {
        "actor": "[/api/v2/users/1/](http://localhost:8013/api/v2/users/1/)"
    },
    "summary_fields": {
        "actor": {
            "id": 1,
            "username": "admin",
            "first_name": "",
            "last_name": ""
        }
    },
    "timestamp": "2024-11-18T13:17:56.321841Z",
    "operation": "create",
    "changes": {
        "name": "cmeyers app",
        "description": "",
        "client_type": "public",
        "redirect_uris": "",
        "authorization_grant_type": "password",
        "skip_authorization": false,
        "organization": "Default-1",
        "id": 1
    },
    "object1": "o_auth2_application",
    "object2": "",
    "object_association": "",
    "action_node": "awx-1",
    "object_type": ""
}
```

</details>

<details><summary>Token Creation - Before</summary>

```
{
    "id": 22,
    "type": "activity_stream",
    "url": "[/api/v2/activity_stream/22/](http://localhost:8013/api/v2/activity_stream/22/)",
    "related": {
        "actor": "[/api/v2/users/1/](http://localhost:8013/api/v2/users/1/)",
        "o_auth2_access_token": [
            "[/api/v2/tokens/1/](http://localhost:8013/api/v2/tokens/1/)"
        ]
    },
    "summary_fields": {
        "o_auth2_access_token": [
            {
                "id": 1,
                "user_id": 1,
                "description": "hello world 2",
                "application_id": 1,
                "scope": "read"
            }
        ],
        "actor": {
            "id": 1,
            "username": "admin",
            "first_name": "",
            "last_name": ""
        }
    },
    "timestamp": "2024-11-18T13:18:22.438100Z",
    "operation": "create",
    "changes": {
        "description": "hello world",
        "application": "cmeyers app",
        "scope": "write",
        "id": 1,
        "token": "************"
    },
    "object1": "o_auth2_access_token",
    "object2": "",
    "object_association": "",
    "action_node": "awx-1",
    "object_type": ""
}
```

</details>

<details><summary>Token Creation - After</summary>

```
{
    "id": 22,
    "type": "activity_stream",
    "url": "[/api/v2/activity_stream/22/](http://localhost:8013/api/v2/activity_stream/22/)",
    "related": {
        "actor": "[/api/v2/users/1/](http://localhost:8013/api/v2/users/1/)"
    },
    "summary_fields": {
        "actor": {
            "id": 1,
            "username": "admin",
            "first_name": "",
            "last_name": ""
        }
    },
    "timestamp": "2024-11-18T13:18:22.438100Z",
    "operation": "create",
    "changes": {
        "description": "hello world",
        "application": "cmeyers app",
        "scope": "write",
        "id": 1,
        "token": "************"
    },
    "object1": "o_auth2_access_token",
    "object2": "",
    "object_association": "",
    "action_node": "awx-1",
    "object_type": ""
}
```
</details>

<details><summary>Application Edit - Before</summary>

```
{
    "id": 23,
    "type": "activity_stream",
    "url": "[/api/v2/activity_stream/23/](http://localhost:8013/api/v2/activity_stream/23/)",
    "related": {
        "actor": "[/api/v2/users/1/](http://localhost:8013/api/v2/users/1/)",
        "o_auth2_access_token": [
            "[/api/v2/tokens/1/](http://localhost:8013/api/v2/tokens/1/)"
        ]
    },
    "summary_fields": {
        "o_auth2_access_token": [
            {
                "id": 1,
                "user_id": 1,
                "description": "hello world 2",
                "application_id": 1,
                "scope": "read"
            }
        ],
        "actor": {
            "id": 1,
            "username": "admin",
            "first_name": "",
            "last_name": ""
        }
    },
    "timestamp": "2024-11-18T13:18:32.145751Z",
    "operation": "update",
    "changes": {
        "description": [
            "hello world",
            "hello world 2"
        ],
        "scope": [
            "write",
            "read"
        ]
    },
    "object1": "o_auth2_access_token",
    "object2": "",
    "object_association": "",
    "action_node": "awx-1",
    "object_type": ""
}
```

</details>

<details><summary>Application Edit - After</summary>

```
{
    "id": 23,
    "type": "activity_stream",
    "url": "[/api/v2/activity_stream/23/](http://localhost:8013/api/v2/activity_stream/23/)",
    "related": {
        "actor": "[/api/v2/users/1/](http://localhost:8013/api/v2/users/1/)"
    },
    "summary_fields": {
        "actor": {
            "id": 1,
            "username": "admin",
            "first_name": "",
            "last_name": ""
        }
    },
    "timestamp": "2024-11-18T13:18:32.145751Z",
    "operation": "update",
    "changes": {
        "description": [
            "hello world",
            "hello world 2"
        ],
        "scope": [
            "write",
            "read"
        ]
    },
    "object1": "o_auth2_access_token",
    "object2": "",
    "object_association": "",
    "action_node": "awx-1",
    "object_type": ""
}
```

</details>

Token edit before/after is also effected in the same way.

### Thoughts

The migrations in this PR change the activity stream entries. Specifically, the deletion of the application and/or token can cause the `summary_fields` section to change.

We recommend that customers and users hook up external logging to their splunk/grafana-thing/elk to track these sorts of things long-term, so I _guess_ it's OK that our activity stream entries change.

**Review by chrismeyersfsu — APPROVED** [review]

**Comment by AlanCoding** [comment]
Unblocked now, because https://github.com/ansible/awx/pull/15647 is merged

File: awx/main/migrations/0199_alter_oauth2application_unique_together_and_more.py [hunk 1]
```diff
@@ -0,0 +1,39 @@
+# Generated by Django 4.2.10 on 2024-10-24 14:06
```
Comment (AlanCoding) [inline awx/main/migrations/0199_alter_oauth2application_unique_together_and_more.py:1]: This migration file should be renamed to the obvious like "remove oauth2 models"

**Review by AlanCoding — CHANGES_REQUESTED** [review]
The attribute `oauth_scopes` is applied to the `user` object. You removed the line where that `setattr` was done, but it appears that 2 cases where this is referenced are not yet removed, in `awx/api/generics.py` and `awx/main/access.py`. Without tokens, the requesting scope should no longer matter, and should be removed

**Comment by PabloHiro** [comment]
Closing this one as a duplicate with conflicts resolved + small patches has been merged already here:
https://github.com/ansible/awx/pull/15666

**Pull request closed without merging**